qemu/tests/block_kill_reconnect_with_remote_storage.py

```diff
@@ -0,0 +1,157 @@
+        image_params = self.params.object_params(
+            self.params.get("local_image_tag"))
+        local_image = self.source_disk_define_by_params(
+            image_params, self.params.get("local_image_tag"))
+        local_image.create(image_params)
+        self.trash.append(local_image)
+
+    def export_local_image_with_nbd(self):
+        self.nbd_export = QemuNBDExportImage(
+            self.params, self.params.get("local_image_tag"))
+        self.nbd_export.export_image()
+
+    def get_disk_storage_name(self, system_disk_cmd, data_disk_cmd):
+        """
+        get data disk name
+        return: data disk name e.g. /dev/sdb
+        """
+        error_context.context("Identify data disk.", logging.info)
+        logging.info('Identify data disk')
+        session = self.vm.wait_for_login(timeout=self.timeout)
+        system_disk_name = session.cmd(system_disk_cmd,
+                                       timeout=self.disk_op_timeout).strip()
+        find_disk_cmd = data_disk_cmd % system_disk_name
+        data_disk_name = session.cmd(find_disk_cmd,
+                                     timeout=self.disk_op_timeout).strip()
+        logging.info('The data disk is %s' % data_disk_name)
+        session.close()
+        return system_disk_name, data_disk_name
+
+    def run_io_test(self, test_disk):
+        """ Run io test on given disks. """
+        error_context.context(
+            "Run io test on %s." % test_disk, logging.info)
+        session = self.vm.wait_for_login(timeout=self.timeout)
+        test_cmd = self.disk_op_cmd % (test_disk, test_disk)
+        session.cmd(test_cmd, timeout=self.disk_op_timeout)
+        session.close()
+
+    def run_iptables(self, cmd):
+        result = process.run(cmd, ignore_status=True, shell=True)
+        if result.exit_status != 0:
+            logging.error('command error: %s' % result.stderr.decode())
+
+    def break_net_with_iptables(self):
+        self.run_iptables(self.params.get('net_break_cmd'))
+        self.net_down = True
+
+    def resume_net_with_iptables(self):
+        self.run_iptables(self.params.get('net_resume_cmd'))
+        self.net_down = False
+
+    def reconnect_loop_io(self):
+        error_context.context(
+            "Run IO test when in reconnecting loop", logging.info)
+        for iteration in range(self.repeat_times):
+            error_context.context("Wait %s seconds" % self.reconnect_time_wait,
+                                  logging.info)
+            time.sleep(self.reconnect_time_wait)
+            self.run_io_test("test_file")
+
+    def check_data_disk_resume(self, test_disk):
+        error_context.context(
+            "check data disk resumed", logging.info)
+        for iteration in range(self.repeat_times):
+            logging.info("Wait %s seconds" % self.reconnect_time_wait)
+            time.sleep(self.reconnect_time_wait)
+            self.run_io_test(test_disk)
+
+    def clean_images(self, net_status):
```

Comment:
passing an argument here might not a good idea, clean_images is called in post_test, which will be called after testing. use self.net_down to check net status

Reply:
thanks, remove passing an argument.